File: pitluga/src/problem1.clj
(defn multiple-of-3? [n]
  (= (mod n 3) 0))
  
(defn multiple-of-5? [n]
  (= (mod n 5) 0))

(defn multiple-of-3-or-5? [n]
  (or (multiple-of-3? n) (multiple-of-5? n)))

(defn sum [coll]
  (reduce + coll))

(defn solve [max]
  (sum (filter multiple-of-3-or-5? (range 1 max))))

File: qxjit/Problem1.hs
module Problem1 where

import Prompt
import System.Environment

multiplesSum :: Integral a => [a] -> [a] -> a
multiplesSum ds ns = sum $ multiples ds ns

multiples :: Integral a => [a] -> [a] -> [a]
multiples ds ns = [n | n <- ns, any (multiple n) ds]

multiple :: Integral a => a -> a -> Bool
multiple n m = n `mod` m == 0

main = do
  putStrLn "Add all the natural numbers below one <max> that are multiples of <ints>"
  max <- prompt "max"
  ds <- promptList "ints"
  putStrLn . show $ (multiplesSum ds [1..(max - 1)])
